keylime/src/crypto.rs

```diff
@@ -1471,22 +1342,27 @@ mod tests {
         assert_eq!(hex, "db9b1cd3262dee37756a09b9064973589847caa8e53d31a9d142ea2701b1b28abd97838bb9a27068ba305dc8d04a45a1fcf079de54d607666996b3cc54f6b67c");
     }
 
-    fn test_x509(privkey: PKey<Private>, pubkey: PKey<Public>) {
+    fn test_x509(privkey: &PKey<Private>, pubkey: PKey<Public>) {
         let tempdir = tempfile::tempdir().unwrap(); //#[allow_ci]
 
-        let r = generate_x509(&privkey, "uuidA", None);
+        let r = CertificateBuilder::new()
+            .private_key(privkey)
+            .common_name("uuidA")
+            .build();
+
         assert!(r.is_ok());
         let cert_a = r.unwrap(); //#[allow_ci]
         let cert_a_path = tempdir.path().join("cert_a.pem");
         let r = write_x509(&cert_a, &cert_a_path);
         assert!(r.is_ok());
         assert!(cert_a_path.exists());
 
-        let r = generate_x509(
-            &privkey,
-            "uuidB",
-            Some(vec!["1.2.3.4".to_string(), "1.2.3.5".to_string()]),
-        );
+        let r = CertificateBuilder::new()
+            .private_key(privkey)
+            .common_name("uuidB")
+            .add_ips(vec!["1.2.3.4".to_string()])
```

Comment:
Why has 1.2.3.5 address been removed in this call? The idea is to test with more than one IP in the vector. Apart from that, I recommend adding also a test with a list of IPv6 tests, if possible

Reply:
Thanks for the review! I moved the unit tests to `keylime/crypto/x509.rs`. I also added a test using IPv6 there.